toolkit/scripts/pkggen.mk

```diff
@@ -202,7 +202,13 @@ ifeq ($(PRECACHE),y)
 $(cached_file): $(STATUS_FLAGS_DIR)/precache.flag
 endif
 
-$(cached_file): $(graph_file) $(go-graphpkgfetcher) $(chroot_worker) $(pkggen_local_repo) $(depend_REPO_LIST) $(REPO_LIST) $(cached_remote_rpms) $(TOOLCHAIN_MANIFEST) $(toolchain_rpms) $(depend_EXTRA_BUILD_LAYERS) $(depend_REPO_SNAPSHOT_TIME) $(STATUS_FLAGS_DIR)/build_packages_cache_cleanup.flag
+ifneq ($(REPO_SNAPSHOT_TIME),)
```

Comment:
The idea was that the `$(depend_REPO_SNAPSHOT_TIME)` would detect changes, if this is running every time the tools run something else is broken. It should be smart enough to only clear the cache when the actual value of the timestamp changes.

Reply:
We might need to add `$(STATUS_FLAGS_DIR)/build_packages_cache_cleanup.flag` as a dependency to the hydrate target.

Reply:
Discussed offline, added a bit more logic and clarification to the comments.